internal/grpc/services/sharesstorageprovider/sharesstorageprovider.go

```diff
@@ -360,8 +359,9 @@ func (s *service) ListStorageSpaces(ctx context.Context, req *provider.ListStora
 	}
 
 	var receivedShares []*collaboration.ReceivedShare
+	var shareEtags map[string]string
 	if fetchShares {
-		lsRes, err := s.sharesProviderClient.ListReceivedShares(ctx, &collaboration.ListReceivedSharesRequest{
+		lsRes, err := s.gateway.ListReceivedShares(ctx, &collaboration.ListReceivedSharesRequest{
```

Comment:
This change will cause a worse performance. Before it we had only one call from sharesstorageprovider to `sharesProvider`. Now it will call gateway inbetween which will then call `sharesProvider`. I understand why this change is necessary but I am a little concerned about performance. 

Should we cache `ListReceivedShares` calls on gateway?

Reply:
> I understand why this change is necessary but I am a little concerned about performance.

Yeah, I understand but let's implement the stuff first now and care about performance optimization later (in the beta phase).

Reply:
Yes sounds good :+1: